frontend/src/features/undo_redo/store/MapHistoryStore.ts

```diff
@@ -122,6 +140,36 @@ function handleAddObjectAction(state: MapState, action: ObjectAddAction): MapSta
   };
 }
 
+function handlePasteClipboardAction(state: MapState, clipboard: MapState): MapState {
+  const newLayers = { ...state.layers };
+
+  clipboard.layers.plant.objects.forEach((object) => {
```

Comment:
Does copy and paste only work for the plant layer?
If so I would also make the action code more specific to the plant layer.
Otherwise here, additional logic for the other layers is needed.